### PR TITLE
feat: add cargo build config and prebuilt support

### PR DESCRIPTION
**Pull request by ecklf**
This PR implements support for correctly resolving custom target build artifacts and adds the relevant test and README section.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,9 +2,18 @@
...
 members = [
     "vercel_runtime",
-    "examples/*",
+    "examples/simple",
+    "examples/cron",
+    "examples/nextjs",
 ]
 
 exclude = [
-    "test/*",
```
Comment (ecklf) [inline Cargo.toml:9]: This was buggy with rust-analyzer, so dropping the glob again.

Comment (dglsparsons) [reply Cargo.toml:9]: Should possibly have been `test/fixtures/*` ?

File: README.md [hunk 2]
```diff
@@ -99,6 +99,26 @@ This Builder supports installing dependencies defined in the `Cargo.toml` file.
...
+For creating a prebuilt deployment, you can create a Cargo build configuration in `.cargo/config.toml`.
+
+```toml
+[build]
+target = "x86_64-unknown-linux-musl"
+
+# Uncomment below to support Rust cross-compilation from macOS to Linux
+# Follow these installation instructions: https://github.com/chinedufn/cross-compile-rust-from-mac-to-linux
+# [target.x86_64-unknown-linux-musl]
+# linker = "x86_64-unknown-linux-gnu-gcc"
```
Comment (dglsparsons) [inline README.md:113]: Is it poss to recommend cargo-zigbuild or cargo-lambda instead?

Comment (dglsparsons) [reply README.md:113]: Oh i see, it's overriding the linker, not the toolchain. 🤔

File: test/fixtures/02-with-utility/vercel.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "functions": {
     "api/**/*.rs": {
-      "runtime": "vercel-rust@4.0.0-beta.3"
+      "runtime": "vercel-dev-builder@0.0.5"
```
Comment (dglsparsons) [inline test/fixtures/02-with-utility/vercel.json:4]: is this intended?

File: test/fixtures/07-with-cargo-configuration/build.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+cargo config -Z unstable-options get target > static/cargo_config.txt
```
Comment (dglsparsons) [inline test/fixtures/07-with-cargo-configuration/build.sh:1]: this looks unrelated to this fixture?

File: README.md [hunk 5]
```diff
@@ -99,6 +99,26 @@ This Builder supports installing dependencies defined in the `Cargo.toml` file.
 
 Furthermore, more system dependencies can be installed at build time with the presence of a shell `build.sh` file in the same directory as the entrypoint file.
 
+## Prebuilt Deployments
+
+For creating a prebuilt deployment, you can create a Cargo build configuration in `.cargo/config.toml`.
```
Comment (dglsparsons) [inline README.md]:
```suggestion
When creating a prebuilt deployment, the build output must be for x86_64 linux. To do this, create a Cargo build configuration at `.cargo/config.toml` with the following contents:
```

Comment (dglsparsons) [reply README.md]: or smth, idk.

File: test/fixtures/02-with-utility/vercel.json (hunk 3 above)
Comment (ecklf) [reply test/fixtures/02-with-utility/vercel.json:4]: yes, all tests now rely on the dev builder to avoid publishing to npm. Just forgot to change for that test.

File: README.md (hunk 2 above)
Comment (ecklf) [reply README.md:113]: Hm we'd need to allow overwriting the build command in the npm module then as we'd need to generate a CLI-compatible build output.

File: test/fixtures/07-with-cargo-configuration/build.sh (hunk 4 above)
Comment (ecklf) [reply test/fixtures/07-with-cargo-configuration/build.sh:1]: This writes what we have defined in `.cargo/config.toml`, so it's intended.

File: Cargo.toml (hunk 1 above)
Comment (ecklf) [reply Cargo.toml:9]: Also tried that, kept complaining about it:
CleanShot 2023-04-05 at 10 01 57

**Review by dglsparsons — APPROVED** [review]
